### ISSUE TITLE
Should histograms return the next key after the last bucket?

### ISSUE THREAD

**Issue by jpountz** [elastic/elasticsearch#28479]
Follow-up to a `discuss` issue: https://discuss.elastic.co/t/to-get-end-date-of-week-in-date-histogram/116591/2

Displaying the value range that each histogram bar maps to can be problematic for the last bar of the histogram, since the histogram doesn't give the key that the next bucket would have. Is it something we should look into adding to the response?

**Comment by DaveCTurner** [comment]
I feel it'd be useful for each bucket to describe both of its bounds. The alternative is that you must assume that the upper bound of each bucket is the lower bound of the next one in the list, which needs a special case for the last bucket (as mentioned here), means that you must preserve the order of the list of buckets, and means you can't simply iterate through the buckets if you're doing some further calculations.

I'd expect there to be code out there that gets a bunch of day-long buckets and incorrectly assumes they're all 84600 seconds long, because that's way easier than trying to get the start time of the next bucket in the loop.

**Comment by jpountz** [comment]
Discussed in FixitFriday: we agreed to put the `from` (inclusive) and `to` (exclusive) bounds of each bucket in the response.

**Comment by justinwyer** [comment]
Putting up my hand for this. If there is any additional context or initial thoughts please share.

**Comment by justinwyer** [comment]
> Discussed in FixitFriday: we agreed to put the from (inclusive) and to (exclusive) bounds of each bucket in the response.

@jpountz to clarify would we be adding a from and to field to each bucket in the response?

**Comment by jpountz** [comment]
@justinwyer Yes.

**Comment by markharwood** [comment]
cc @elastic/es-search-aggs

**Comment by ankitabhatt17** [comment]
Is this feature is implemented in current verison?

@markharwood

**Comment by polyfractal** [comment]
@ankitabhatt17 No, this has not been implemented. If this enhancement is added, this ticket will be updated/linked to the appropriate PR (or closed if we decide not to implement it).
